bindata/network/ovn-kubernetes/ovnkube-master.yaml

```diff
@@ -311,8 +311,11 @@ spec:
                   return
             fi
             echo "Checking ${db} health"
-            serverid=$(ovsdb-tool show-log "${db}" | sed -ne '/server_id:/s/server_id: *\([[:xdigit:]]\+\)/\1/p')
-            match=$(ovsdb-tool show-log "${db}" | grep "prev_servers: *${serverid}(" || true)
+            # remove leading and trailing space around server_id
+            serverid=$(ovsdb-tool show-log "${db}" | sed -ne '/server_id:/s/ *server_id: *\([[:xdigit:]]\+\).*/\1/p')
```

Comment:
The sed can be simplified to just

        sed -ne 's/ *server_id: *\([[:xdigit:]]\+\).*/\1/p'

The `s///` rule already won't match if there's no "`server_id:`" in the string, so the extra `/server_id:/` at the front wasn't doing anything

Reply:
I was originally matching the original grep plus sed formulation and the simpler regex `/server_id:/` would be faster to match that the substitution regex but it won't matter much.  

I finally check the ovsdb-tools source and there is a fixed amount of space after the `server_id: ` so we can use a positive lookbehind with `grep -o` and that should simplify things.  Also threw in a `-m 1` to exit after the first match for good measure, assuming that there should always be only one match.  The `-m 1` causes `ovsdb-tool` to complain about the broken pipe but that is okay since it's on stderr.

`grep -m 1 -oP '(?<=server_id: )[[:xdigit:]]+'`